### PR TITLE
PR #4: Add comprehensive tests for cluster name support

### PR DESCRIPTION
**Pull request by gulati-aakriti**
## Description

Adds tests for cluster-support changes

Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

Please describe the tests that were run to verify your changes and steps to reproduce. Please specify any test configuration required. 

- [x] New Test X
- [ ] Functional testsuite

**Test Configuration**
* Kubernetes clusters tested on: 

## Checklist :dart:

- [x] Followed coding guidelines
- [x] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [x] Tests added or updated

## Additional information

Include any additional information such as links, test results, screenshots here

## Summary by Sourcery

Add comprehensive validation coverage for Bulk API cluster name handling and model settings, including backend passthrough and datasource cluster list behavior.

Tests:
- Extend Bulk API integration tests to cover cluster_name and model_settings validation scenarios, including success, failure, and edge cases.
- Add unit tests for BulkJobManager to verify cluster name passthrough and backward compatibility when cluster name is absent.
- Add unit tests for KruizeDataSourceEntry to validate clusters JSONB parsing, filtering of invalid names, and behavior when clusters are missing.
- Add unit tests for BulkServiceValidation to cover cluster_name validation rules, trimming, length limits, and optional null handling.

### CODE REVIEW

File: tests/scripts/local_monitoring_tests/rest_apis/test_bulkAPI.py [hunk 1]
```diff
@@ -377,3 +377,70 @@ def test_bulk_api_filter_application(
...
+@pytest.mark.test_bulk_api_ros
+@pytest.mark.parametrize("cluster_name, expected_status, expected_error", [
+    ("prod-cluster", SUCCESS_200_STATUS_CODE, None),  # Valid cluster name
+    ("cluster-01", SUCCESS_200_STATUS_CODE, None),  # Valid with numbers
+    ("cluster.us-east.prod", SUCCESS_200_STATUS_CODE, None),  # Valid with dots
+    ("  prod-cluster  ", SUCCESS_200_STATUS_CODE, None),  # Valid with whitespace (should be trimmed)
+    ("", ERROR_STATUS_CODE, "cluster_name cannot be an empty string"),  # Empty string
+    ("Cluster-A", ERROR_STATUS_CODE, "Invalid cluster_name format"),  # Uppercase
+    ("-cluster", ERROR_STATUS_CODE, "Invalid cluster_name format"),  # Starts with hyphen
+    ("cluster_name", ERROR_STATUS_CODE, "Invalid cluster_name format"),  # Underscore
```
[the remark this answers is not in the document]
Comment (gulati-aakriti) [reply tests/scripts/local_monitoring_tests/rest_apis/test_bulkAPI.py]: resolved in pr1 and pr2

File: src/test/java/com/autotune/database/table/KruizeDataSourceEntryTest.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    @Test
+    @DisplayName("Test setting and getting cluster list with multiple clusters")
+    public void testSetAndGetClusterList() {
+        KruizeDataSourceEntry entry = new KruizeDataSourceEntry();
+        List<String> clusters = Arrays.asList("default", "stage", "prod");
+        
+        entry.setClusterList(clusters);
+        
+        assertEquals("default,stage,prod", entry.getClusters());
+        assertEquals(clusters, entry.getClusterList());
```
[the remark this answers is not in the document]
Comment (gulati-aakriti) [reply src/test/java/com/autotune/database/table/KruizeDataSourceEntryTest.java]: added

**Comment by khansaad** [comment]
@sourcery-ai review

**Review by bharathappali — APPROVED** [review]
LGTM

**Review by kusumachalasani — APPROVED** [review]
lgtm!
